### PR TITLE
RGB gamer lights

### PR DESCRIPTION
**Pull request by SonixApache**
Allows you to add wired glass tiles to light fixtures to customize the light colors with a multitool.

prompt 
lights

if you know of any more relevant issues or feature requests that can be closed, lemme know and I'll add them in here

:cl:
 * experiment: You can now customize lights! Add some wired glass to a light fixture, then use a multitool on it.

### CODE REVIEW

**Comment by Pastamadness** [comment]
Red LEDs should heat the room up slowly and blue should cool it for realism.

**Comment by DeityLink** [comment]
not a fan of the plastic+multitool steps. You could set the color by adding plastic (suggesting that you're using colored plastic) and then a multitool to limit the range/power.
If you want RGB tubes you should add them as a separate items just like high efficiency tubes are.

Also don't close the issue, it suggested remote controlled lights which is a really cool idea.

**Comment by SonixApache** [comment]
>If you want RGB tubes you should add them as a separate items just like high efficiency tubes are.

Though I agree that it's a bit too easy, this is a mostly QoL thing and I think the step of having to add plastic to all your lights is bad enough as-is. Even more light tubes to choose from just seems needless.

> Also don't close the issue, it suggested remote controlled lights which is a really cool idea.

okay

**Comment by brndd** [comment]
I feel like painting lights would be a nicer solution than multitooling them.

**Comment by DeityLink** [comment]
> I feel like painting lights would be a nicer solution than multitooling them.

That's already something that'll come with the BEEG PAINT PR in progress. But I can always do it so that the lights combine or something. They're not mutually exclusive.

**Comment by Eneocho** [comment]
Clearly the solution is to inject chems into the lights so the vapors change how they glow.
Mercury for a blue (something something spess tubes magic neon that only manifests when injected with chems). Hydrogen for a bright red, helium for yellow, carbon and oxygen for white.
Neonlights when??????

**Comment by jellyveggie2** [comment]
>Should I add more steps to construction, or light tube subtypes?

Something like Ancientpower's smart lights (#24959), maybe? (For the light tube subtypes I mean)

**Comment by NigglyWiggly** [comment]
> Clearly the solution is to inject chems into the lights so the vapors change how they glow.
> Mercury for a blue (something something spess tubes magic neon that only manifests when injected with chems). Hydrogen for a bright red, helium for yellow, carbon and oxygen for white.
> Neonlights when??????

Injecting plasma into every light on the station so that all the lights can be a cool purple color

**Comment by SonixApache** [comment]
~~i'll work on implementing the stuff from #24959 and moving the construction steps from the light fixture to the tube itself, but for now this'll stay as-is just in case i ran out of codermana halfway through~~
i hate #24959's implementation and don't want to touch it whatsoever

**Comment by jellyveggie2** [comment]
I wanted to mess around with this a bit and came across a strange bug: Placing a light tube in a light fixture causes you to hit the tube with itself, breaking it more often than not.

**Comment by pedr69** [comment]
LED tubes should be autolathe illegal or something
if they were legal to use why doesn't nanotrasen use them by default to save 70% the stations power consumption?

**Comment by Eneocho** [comment]
> LED tubes should be autolathe illegal or something if they were legal to use why doesn't nanotrasen use them by default to save 70% the stations power consumption?

That's what the high efficiency lights are.
Actually, requesting that instead of adding a new light type, we add this functionality to the high efficiency lights.

**Comment by DeityLink** [comment]
So @ancientpower actually added smart rgb lights 4 years ago and they stand unobtainable and unused in the code. You should use those instead. #24959

**Comment by xXxEMBERGHOSTSQUADDIExXx** [comment]
> Clearly the solution is to inject chems into the lights so the vapors change how they glow.
Mercury for a blue (something something spess tubes magic neon that only manifests when injected with chems). Hydrogen for a bright red, helium for yellow, carbon and oxygen for white.
Neonlights when??????

something like this would be cool for fires too because the dynamic fire color added in #35127 uses blackbody color but wood burning is orange because of sodium apparently and also u could have a custom firecolor for plasma like you mentioned @Eneocho

right now theres recycling materials and construction materials, it might be good to decide on a standard concept of what an object is made of materially so it can be used in different places like hitsound, fragmentation, flame color, and stuff
there are material datum for ores nd stuff, maybe it could be merged with that

File: code/modules/power/lighting/lighting.dm [hunk 1]
```diff
@@ -302,8 +324,52 @@ var/global/list/obj/machinery/light/alllights = list()
 // attack with item - insert light (if right type), otherwise try to break the light
 
 /obj/machinery/light/attackby(obj/item/W, mob/living/user)
-	user.delayNextAttack(8)
+	if(ismultitool(W)) //RGB gamer lights
+		if(!rgb_upgrade)
+			to_chat(user, "There's nothing to connect \the [W] to on \the [src].")
+			return
+		var/which = alert(user, "What would you like to change?", "RGB Gamer lights", "Color", "Power", "Range")
```
Comment (nervere) [inline code/modules/power/lighting/lighting.dm]: Please do not call them "gamer lights" in an IC text box.

Comment (xXxEMBERGHOSTSQUADDIExXx) [reply code/modules/power/lighting/lighting.dm]: soulless

Comment (SonixApache) [reply code/modules/power/lighting/lighting.dm]: ok

**Comment by pedr69** [comment]
we seem to have LED lights already?
normal ones are tungsten, HE are halogen and then there's an unused and unsettable LED smartlight
https://github.com/vgstation-coders/vgstation13/blob/65af56430968cb6b28fe6878c62903be346e45f8/code/modules/power/lighting/lighting.dm#L609-L610
https://github.com/vgstation-coders/vgstation13/blob/65af56430968cb6b28fe6878c62903be346e45f8/code/modules/power/lighting/lighting.dm#L622-L623
https://github.com/vgstation-coders/vgstation13/blob/65af56430968cb6b28fe6878c62903be346e45f8/code/modules/power/lighting/lighting.dm#L632-L633

**Comment by jwhitak** [comment]
Plastic then multitool is the laziest creation method. Use something else imo. At least paint the plastic? Use light tiles instead?

**Comment by Eneocho** [comment]
I still think this should be a feature of HE lights rather than their own thing.

**Comment by SonixApache** [comment]
add wired glass to the HE lights and it will be a HE light feature

**Comment by pedr69** [comment]
so are you gonna move the colors to the RGBtubes or not?
if not, you should update the PR and changelog to say "light fixtures" instead of "any light"

**Comment by SonixApache** [comment]
> so are you gonna move the colors to the RGBtubes or not? if not, you should update the PR and changelog to say "light fixtures" instead of "any light"

:cl: updated

**Comment by ShiftyRail** [comment]
Europa lights would make this suck less.

**Comment by xXxEMBERGHOSTSQUADDIExXx** [comment]
when

**Pull request merged**